### DISCUSSION TITLE
Unable to retrieve variable from secrets backend (SecretsManagerBackend).

### DISCUSSION THREAD

**Discussion by rankol** [apache/airflow#23879] [Q&A] [answered]
Hello, I am using MWAA (AWS Managed Airflow) and am new to writing Python/DAGs. Currently I am coming across some errors when attempting to access a secret in AWS Secrets Manager.

Here is some code I copied for testing to access a variable in Secrets Manager.
[https://gist.github.com/rankol/58faedb35a6a07b2023b99e7fa7d0862](https://gist.github.com/rankol/58faedb35a6a07b2023b99e7fa7d0862)

Documentation that I attempted to follow.
[https://docs.aws.amazon.com/mwaa/latest/userguide/samples-secrets-manager-var.html](https://docs.aws.amazon.com/mwaa/latest/userguide/samples-secrets-manager-var.html)

Here is the error message when attempting to access a variable. [https://gist.github.com/rankol/146d3ec945ea03fe0bf556c66e064360](https://gist.github.com/rankol/146d3ec945ea03fe0bf556c66e064360)

Here is some code I was wrote for testing access to a regularly named secret that does not use any prefixing such as `airflow/connections` or `airflow/variables`. [https://gist.github.com/rankol/2bf7cfc3de22f3f65875a6d4c82850f4](https://gist.github.com/rankol/2bf7cfc3de22f3f65875a6d4c82850f4)

The error message is similar to the variable error message the only difference is that I see the text `ERROR - Unable to retrieve connection from secrets backend` instead of `variable`.

My MWAA execution role has permissions to access the secret but I have not declared any credentials for the IAM role within the DAG. I at least am under the assumption that credentials from the IAM role are able to be used by AwsBaseHook.

Here is the airflow configuration setup for the backend taken from the AWS console on my side. This is based off documentation from [here](https://docs.aws.amazon.com/mwaa/latest/userguide/connections-secrets-manager.html). One thing to note is that the map below has single quotes while documentation uses double quotes.
```
secrets.backend : airflow.providers.amazon.aws.secrets.secrets_manager.SecretsManagerBackend
secrets.backend_kwargs : {'connections_prefix' : 'airflow/connections', 'variables_prefix' : 'airflow/variables'}
```
I am unsure of what the error is here. My secret name does NOT use `airflow/variables` or `airflow/connections` in the naming scheme. Is this what the main issue is or is this more of a connectivity or IAM role credentials issue? If this is just due to the naming scheme not matching the secrets.backend_kwargs is there a proper way to access a regularly named secret?

**Comment by potiuk** [comment, marked as answer]
I cannot comment on the correctness but the error suggests you are using wrong protocol - usually you get this kind of error when you try SSL over plain connection (without SSL) or when you connect to an endpoint that speaks different protocol (so for example when you connect to an HTTP server serving UI, where what you really want is to connect to a port that serves another, non-HTTP API) . 

Another reason could be that you have some kind of firewall between you and the server and that firewall breaks the connection during establishing - because it has rules to tells it so.

Not sure if that's helpful but you should look at those potential issues.

**Reply by rankol** [reply]
@potiuk Thanks for the reply. This was a firewall issue on our end. We setup a VPC endpoint for Secrets Manager and I no longer get the `Connection was closed` error message.

**Reply by potiuk** [reply]
Thanks for letting us know!
